src/addons/blog/services/blog.ts

```diff
@@ -134,10 +208,28 @@ export class AddonBlogProvider {
      * @returns Entry deleted successfully or not.
      * @since 4.4
      */
-    async deleteEntry(params: AddonBlogDeleteEntryWSParams, siteId?: string): Promise<AddonBlogDeleteEntryWSResponse> {
+    async deleteEntry(params: AddonBlogDeleteEntryWSParams, siteId?: string): Promise<void> {
         const site = await CoreSites.getSite(siteId);
 
-        return await site.write('core_blog_delete_entry', params);
+        try {
+            if (!CoreNetwork.isOnline()) {
+                return await AddonBlogOffline.markEntryAsRemoved(params.entryid, siteId);
+            }
+
+            await site.write('core_blog_delete_entry', params);
+            const isMarkAsRemoved = await AddonBlogOffline.getEntryToRemove(params.entryid);
+
+            if (isMarkAsRemoved) {
+                await AddonBlogOffline.unmarkEntryAsRemoved(params.entryid);
+            }
```

Comment:
```suggestion
            await CoreUtils.ignoreErrors(AddonBlogOffline.unmarkEntryAsRemoved(params.entryid));
```
Instead of reading and writing, you can just write and ignore the errors. And the `ignoreErrors` can probably be removed too because you delete using a WHERE, so it won't throw an error if no entries are found, but you can leave it just in case.